POC/py/parser.py

```diff
@@ -0,0 +1,306 @@
+import sys
+import re
+import os
+import json
+import asn1tools
+import shutil
+from collections import deque
+
+#accept multiple files
+#check and validate for well formness of asn1 file by compiling input using asn1 tools
+
+# take a file name as arg
+file_path=sys.argv[1]
+
+if file_path.endswith('.asn1'):
+    with open(file_path, 'r') as file:
+        data = file.read()
+        # print(data)
+else:
+    print("Invalid file type")
+    sys.exit(1)
+
+
+
+
+
+# parse the file line by line to find ::= pattern
+# if found, extract the schema name and type
+
+lines=data.splitlines()
+# print(lines)
+
+count=0
+lines_count=len(lines)
+namesspaces=[]
+
+dirs=["../files/schema","../files/schema_updated"]
+
+for dir in dirs:
+    if not os.path.exists(dir):
+        os.makedirs(dir)
+
+    files=os.listdir(dir)
+    for file in files:
+        os.remove(f"{dir}/{file}")
+
+
+"""
+Namespace parser
+"""
+isNamespace=False
+start=0
+end=0
+namespaces={}
+while count<lines_count:
+    line=lines[count]
+    if "::= BEGIN" in line:
+        namespace=line.split()[0].strip()
+        # print("Namespace:",namespace) 
+        isNamespace=True
+        start=count
+    if "END" in line:
+        isNamespace=False
+        end=count
+        end+=1
+        namespaces[namespace]={
+            "start":start,
+            "end":end
+        }
+    
+    count+=1
+
+"""
+    Write each namespace data to a file
+"""
+# print(namesspaces)
+for i in namespaces:
+    with open(f"../files/namespace/{i}.asn1",'w') as file:
+        for j in range(namespaces[i]["start"],namespaces[i]["end"]):
+            file.write(lines[j]+'\n')
+
+
+namespace_files=os.listdir("../files/namespace")
+
+
+"""
+Parse Each file and load into Memory
+"""
+# possible_types=["ENUMERATED","INTEGER","CHOICE","IA5String","CLASS","BIT","SEQUENCE","BOOLEAN","OCTET"]
+# schema_types=set()
+
+def find_between_parentheses(s):
```

Comment:
for each function defined, please add a description introducing purpose of the function, its input and its output